libraries/AP_BattMonitor/AP_BattMonitor.cpp

```diff
@@ -1270,7 +1270,7 @@ void AP_BattMonitor::MPPT_set_powered_state_to_all(const bool power_on)
 // it will supply energy if available.
 void AP_BattMonitor::MPPT_set_powered_state(const uint8_t instance, const bool power_on)
 {
-    if (instance < _num_instances) {
+    if (instance < _num_instances && drivers[instance] != nullptr) {
```

Comment:
```suggestion
    if (instance < ARRAY_SIZE(drivers) && drivers[instance] != nullptr) {
```

... makes it clear that the following deref is valid

Reply:
I'm not sure we should replace _num_instances with ARRAY_SIZE(drivers) because this will make the single call different from the 10+ other similar checks we do in AP_BattMonitor.  Either way may be fine but I think we should be consistent to avoid the possibility of odd bugs creeping in.

BTW the "voltage(uint8_t instance) const" getter is misisng the nullptr check as well it seems